src/test/unit/bqrc_kalman_unittest.cc

```diff
@@ -0,0 +1,197 @@
+
+    biquadFilter_t bqrcFilter;
+    biquadRCFIR2FilterInit(&bqrcFilter, cutoffFrequency, dT);
+
+    double kalmanVec[SAMPLE_SET_SIZE];
+    double bqrcVec[SAMPLE_SET_SIZE];
+    double lastK = 0;
+    int kConvergeSample = 0;
+
+    srand(time(NULL));
+    for (int i = 0; i < SAMPLE_SET_SIZE; i++) {
+        const float randVal = (rand() % 20000) * 1e-1f;
+
+        kalmanVec[i] = fastKalmanUpdate(&fkfFilter, randVal);
+        bqrcVec[i] = biquadFilterApply(&bqrcFilter, randVal);
+
+        if (fkfFilter.k > lastK) {
+            lastK = fkfFilter.k;
+        } else if (kConvergeSample == 0) {
+            kConvergeSample = i;
+        }
+    }
+
+    const int numSamples = SAMPLE_SET_SIZE - 1 - kConvergeSample;  // offset for index value
+    const float correlation = corr(kalmanVec, bqrcVec, kConvergeSample, numSamples);
+    const float convergeTimeMs = (kConvergeSample + 1) * dT * 1000;
+    const double fkfMean = mean(kalmanVec, kConvergeSample, numSamples);
+    const double bqrcMean = mean(bqrcVec, kConvergeSample, numSamples);
+    const double meanDifference = abs(fkfMean - bqrcMean);
+    const double fkfDispersion = var(kalmanVec, kConvergeSample, numSamples) / fkfMean;
+    const double bqrcDispersion = var(bqrcVec, kConvergeSample, numSamples) / bqrcMean;
+    const double dispersionDifference = abs(fkfDispersion-bqrcDispersion);
+
+    printf("\n[:::] FKF Q : %d\n", q);
+    printf("[:::] FKF R : %d\n", r);
+    printf("[:::] BQRC Cutoff Frequency: %.10f\n", cutoffFrequency);
+    printf("[:::] FKF K converged at sample %d (%.3fms)\n", kConvergeSample + 1, convergeTimeMs);
+    printf("===== Post-transient Statistics (%d samples) =====\n", numSamples);
+    printf("[:::] Correlation : %.10e\n", correlation);
+    printf("[:::] Mean Difference : %.10e\n", meanDifference);
+    printf("[:::] Dispersion Difference: %.10e\n\n", dispersionDifference);
+    EXPECT_LT(PASSING_CORRELATION, correlation);
+}
+
+TEST(BQRCKalmanTest, BQRCKalmanFlightDataCorrelationTest)
+{
+    // Input parameters exposed to pilot
+    const uint16_t q = 400;
+    const uint16_t r = 88;
+    const uint8_t p = 0;
+    const float rate = 32000;
+
+    fastKalman_t fkfFilter;
+    fastKalmanInit(&fkfFilter, q, r, p);
+
+    const float dT = 1 / rate;
+    const float bqrcQ = fkfFilter.q;
+    const float bqrcR = fkfFilter.r;
+    const float bqrcK = 2 * sqrt(bqrcQ) / (sqrt(bqrcQ + 4 * bqrcR) + sqrt(bqrcQ));
+    const float cutoffFrequency = 1 / (2 * M_PI_FLOAT * 0.5f * dT * (1 / bqrcK - 2));
+
+    biquadFilter_t bqrcFilter;
+    biquadRCFIR2FilterInit(&bqrcFilter, cutoffFrequency, dT);
+
+    double kalmanVec[SAMPLE_SET_SIZE];
+    double bqrcVec[SAMPLE_SET_SIZE];
+    double lastK = 0;
+    int kConvergeSample = 0;
+
+    srand(time(NULL));
+    int samplesEvaluated = 0;
+    FILE *file;
+    FILE *out;
+    file = fopen("unit/flightDebugData.csv", "r");
+    out = fopen("unit/flightDebugOut.csv", "w");
+
```

Comment:
Error checking is missing

Reply:
Maybe it's better to include test data as a string inside code? Your opinion?

Reply:
Both are a bit cumbersome ... 
simple assert() or unittest check will handle missing failed case.

Reply:
I don't want to carry around additional files for a test, we could easily provide them in-place.

Reply:
Should perhaps add header to the outputfile for easy interpretting